src/m365/outlook/commands/message/message-remove.ts

```diff
@@ -0,0 +1,142 @@
+import auth from '../../../../Auth.js';
+import { Logger } from '../../../../cli/Logger.js';
+import GlobalOptions from '../../../../GlobalOptions.js';
+import request, { CliRequestOptions } from '../../../../request.js';
+import { accessToken } from '../../../../utils/accessToken.js';
+import GraphCommand from '../../../base/GraphCommand.js';
+import commands from '../../commands.js';
+import { cli } from '../../../../cli/cli.js';
+import { validation } from '../../../../utils/validation.js';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  id: string;
+  userId?: string;
+  userName?: string;
+  force?: boolean
+}
+
+class OutlookMessageRemoveCommand extends GraphCommand {
+  public get name(): string {
+    return commands.MESSAGE_REMOVE;
+  }
+
+  public get description(): string {
+    return 'Removes a specifc message from a mailbox';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        userId: typeof args.options.userId !== 'undefined',
+        userName: typeof args.options.userName !== 'undefined',
+        force: !!args.options.force
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-i, --id <id>'
+      },
+      {
+        option: '--userId [userId]'
+      },
+      {
```

Comment:
Options `id` and `userId` should be validated when specified.

Reply:
Do you know how to validate message id?
`AAMkAGVmMDEzMTM4LTZmYWUtNDdkNC1hMDZiLTU1OGY5OTZhYmY4OABGAAAAAAAiQ8W967B7TKBjgx9rVEURBwAiIsqMbYjsT5e-T7KzowPTAAAAAAEMAAAiIsqMbYjsT5e-T7KzowPTAALvuv07AAA=`
I would prefer server side exception.

Reply:
I have no clue if we can validate this. It looks like a base64 string, but I don't know if there's any documentation what should be inside this string.

Reply:
Yes, message id should contain encoded info about mailbox folder and item inside the folder in exchange, but even if we decode base64 string, we can't validate each part.

Reply:
No indeed, and it's a bit tricky because we don't have any documentation about it. It could be possible that older messages use another format. So therefore I'd say let's not try to validate the id and potentially block users from using this command.

Reply:
I just noticed that I wrote `id` instead of `userId`, sorry for that, this caused a bit of confusion 😊